From f4eaaba74cfbd39472a56fa89f0583369bf6c454 Mon Sep 17 00:00:00 2001
From: rmuxnet <armandas.kvietkus@proton.me>
Date: Wed, 1 Jul 2026 00:31:08 +0200
Subject: [PATCH] mmc: sdhci-pci: enable PS4 Baikal SDHCI quirk entry, was
 commented out
MIME-Version: 1.0
Content-Type: multipart/mixed; boundary="------------false"

This is a multi-part message in MIME format.
--------------false
Content-Type: text/plain; charset=UTF-8; format=fixed
Content-Transfer-Encoding: 8bit

---
 drivers/mmc/host/sdhci-pci-core.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)


--------------false
Content-Type: text/x-patch; name="0113-mmc-sdhci-pci-enable-PS4-Baikal-SDHCI-quirk-entry-wa.patch"
Content-Transfer-Encoding: 8bit
Content-Disposition: attachment; filename="0113-mmc-sdhci-pci-enable-PS4-Baikal-SDHCI-quirk-entry-wa.patch"

diff --git a/drivers/mmc/host/sdhci-pci-core.c b/drivers/mmc/host/sdhci-pci-core.c
index 691255dc172ca2f5a5dcf01196370aa84102f63e..9828e73d242c589af3d4925e6103bf04a2137cfd 100644
--- a/drivers/mmc/host/sdhci-pci-core.c
+++ b/drivers/mmc/host/sdhci-pci-core.c
@@ -372,7 +372,8 @@ static int aeolia_probe_slot(struct sdhci_pci_slot *slot)
 	* using host capabiilites. So it should be harmless in all cases.
 	*/
 	if ((slot->chip->pdev->device == PCI_DEVICE_ID_SONY_AEOLIA_SDHCI) || \
-	    (slot->chip->pdev->device == PCI_DEVICE_ID_SONY_BELIZE_SDHCI)) {
+	    (slot->chip->pdev->device == PCI_DEVICE_ID_SONY_BELIZE_SDHCI) || \
+	    (slot->chip->pdev->device == PCI_DEVICE_ID_SONY_BAIKAL_SDHCI)) {
 		slot->host->quirks2 |= SDHCI_QUIRK2_PRESET_VALUE_BROKEN;
         }
 
@@ -2033,8 +2034,7 @@ static const struct pci_device_id pci_ids[] = {
 	#ifdef CONFIG_X86_PS4
 	SDHCI_PCI_DEVICE(SONY, AEOLIA_SDHCI, aeolia),
 	SDHCI_PCI_DEVICE(SONY, BELIZE_SDHCI, aeolia),
-	// TODO (ps4patches): What is this doing in comments?
-	//SDHCI_PCI_DEVICE(SONY, BAIKAL_SDHCI, aeolia),
+	SDHCI_PCI_DEVICE(SONY, BAIKAL_SDHCI, aeolia),
 	#endif
 	SDHCI_PCI_DEVICE(GLI, 9750, gl9750),
 	SDHCI_PCI_DEVICE(GLI, 9755, gl9755),

--------------false--


